projects/dotcms-field-elements/src/components/dot-date/dot-date.e2e.ts

```diff
@@ -1,131 +1,385 @@
+                element.setProperty('name', 'date');
+                await page.waitForChanges();
+                const label = await dotTestUtil.getDotLabel(page);
+                expect(label.getAttribute('name')).toBe('date');
+            });
+
+            it('should render default value', () => {
+                expect(inputCalendar.getAttribute('name')).toBe('');
             });
-            expect(evt_statusChange.events).toEqual([]);
-            expect(evt_valueChange.events).toEqual([]);
         });
 
-        it('should emit status and value on Reset', async () => {
-            await element.callMethod('reset');
-            expect(spyStatusChangeEvent).toHaveReceivedEventDetail({
-                name: 'date01',
-                status: {
-                    dotPristine: true,
-                    dotTouched: false,
-                    dotValid: false
-                }
+        describe('label', () => {
+            it('should set label prop in dot-label', async () => {
+                element.setProperty('label', 'test');
+                await page.waitForChanges();
+                const label = await dotTestUtil.getDotLabel(page);
+                expect(label.getAttribute('label')).toBe('test');
+            });
+
+            it('should render default value', async () => {
+                const label = await dotTestUtil.getDotLabel(page);
+                expect(label.getAttribute('label')).toBe('');
+            });
+        });
+
+        describe('hint', () => {
+            it('should set hint correctly', async () => {
+                element.setProperty('hint', 'Test');
+                await page.waitForChanges();
+                expect((await dotTestUtil.getHint(page)).innerText).toBe('Test');
+            });
+
+            it('should not render hint', async () => {
+                expect(await dotTestUtil.getHint(page)).toBeNull();
+            });
+        });
+
+        describe('required', () => {
+            it('should render required attribute with invalid value', async () => {
+                element.setProperty('required', { test: 'test' });
+                await page.waitForChanges();
+                expect(inputCalendar.getAttribute('required')).toBeDefined();
+            });
+
+            it('should not render required attribute', async () => {
+                element.setProperty('required', 'false');
+                await page.waitForChanges();
+                expect(inputCalendar.getAttribute('required')).toBeNull();
+            });
+
+            it('should render required attribute for the dot-label', async () => {
+                element.setProperty('required', 'true');
+                await page.waitForChanges();
+                const label = await dotTestUtil.getDotLabel(page);
+                expect(label.getAttribute('required')).toBeDefined();
+            });
+        });
+
+        describe('requiredMessage', () => {
```

Comment:
Here you're not testing that the error show, I can go to the component and remove [this](https://github.com/dotCMS/core-web/blob/f762318d7be8e8378cd8cf5416fa6d21c32521f7/projects/dotcms-field-elements/src/components/dot-date/dot-date.tsx#L144) and the test will keep passing.

In fact, passing the messages as atrr to a child component to then get them back in an event is not a good approach because we're passing down a message that will be shown where is getting passed from, so it doesn't make sense, this was something we all miss in the code review.

The error message can be calculated to show or hide using the `require` prop and the `status` in this case the `dot-input-calendar` don't have to do it itself this is a job of the parent to do that.

Reply:
fixed with the interface, `DotInputCalendarStatusEvent `